design_linked_list.py

```diff
@@ -0,0 +1,73 @@
+class Node:
+    def __init__(self, val = 0, next = None):
+        self.val = val
+        self.next = next
+        
```

Comment:
sending None as a default value than using direclty None inside the code is very good approach.